app/controllers/dor_controller.rb

```diff
@@ -46,10 +46,16 @@ def reindex
     @solr_doc = Argo::Indexer.reindex_pid params[:pid], index_logger
     Dor::SearchService.solr.commit # reindex_pid doesn't commit, but callers of this method may expect the update to be committed immediately
     flash[:notice] = "Successfully updated index for #{params[:pid]}"
-    redirect_to :back
+    render status: 200, text: flash[:notice] and return unless request.headers['Referer']
+    redirect_back(
+      fallback_location: proc { catalog_path(params[:pid])}
+    )
   rescue ActiveFedora::ObjectNotFoundError # => e
     flash[:error] = 'Object does not exist in Fedora.'
-    redirect_to :back
+    render status: 404, text: flash[:error] and return unless request.headers['Referer']
```

Comment:
`and` here is appropriate as it is a Rails thing to use `and return` in controllers.